aptos-move/framework/aptos-stdlib/sources/data_structures/smart_table.move

```diff
@@ -134,6 +142,44 @@ module aptos_std::smart_table {
         }
     }
 
+    /// Add multiple key/value pairs to the smart table. The keys must not already exist.
+    public fun add_all<K, V>(
+        table: &mut SmartTable<K, V>,
+        keys: vector<K>,
+        values: vector<V>,
+    ) {
+        assert!(vector::length(&keys) == vector::length(&values), error::invalid_argument(EMISMATCHED_LENGTHS));
+        vector::zip(keys, values, |key, value| {
+            add(table, key, value);
+        });
+    }
+
+    inline fun unzip_entries<K: copy, V: copy>(entries: &vector<Entry<K, V>>): (vector<K>, vector<V>) {
+        let keys = vector[];
+        let values = vector[];
+        vector::for_each_ref(entries, |e|{
+            let entry: &Entry<K, V> = e;
+            vector::push_back(&mut keys, entry.key);
+            vector::push_back(&mut values, entry.value);
+        });
+        (keys, values)
+    }
+
+    /// Convert a smart table to a simple_map.
+    /// Disclaimer: This function may be costly. Use it at your own discretion.
```

Comment:
I'd expand this a bit more to explain why it can be costly. We can also mention that this should only be used for view functions ideally, which is the most useful use case